op-service/logfilter/logfilter.go

```diff
@@ -0,0 +1,222 @@
+	return func(ctx context.Context, l slog.Level) tri.Tri {
+		return tri.FromBool(l == lvl)
+	}
+}
+
+type Statement func(ctx context.Context, lvl slog.Level) tri.Tri
+
+// Selector to Statement Conversions
+func undefinedLogSelector(ctx context.Context, lvl slog.Level) tri.Tri {
+	return tri.Undefined
+}
+
+func (f Selector) Mute() Statement {
+	// T => F
+	// F => U
+	// U => U
+	return Statement(f.Or(undefinedLogSelector).Not())
+}
+
+func (f Selector) Show() Statement {
+	// T => T
+	// F => U
+	// U => U
+	return Statement(f.Or(undefinedLogSelector))
+}
+
+// Filter Building and Composition
+// buildFilter constructs the core predicate used by DefaultShow / DefaultMute.
+//
+//	defaultShow – baseline decision when every selector returns tri.Undefined
+//	filters     – ordered list of LogFilters (last match wins)
+func buildFilter(defaultShow bool, filters ...Statement) Filter {
+	return func(ctx context.Context, lvl slog.Level) bool {
+		// Walk the slice in reverse so the *last* filter defined by the caller
+		// is evaluated first.  As soon as we get a defined decision we can exit.
+		for i := len(filters) - 1; i >= 0; i-- {
+			if d := filters[i](ctx, lvl); d != tri.Undefined {
+				return d.Bool(defaultShow) // early return — nothing earlier can override
+			}
+		}
+		// No filter made a decision; fall back to the caller's default.
+		return defaultShow
+	}
+}
+
+// DefaultShow creates a new handlerFilter that shows logs that match the given filters.
+// Later filters override earlier ones.
+func DefaultShow(filters ...Statement) Filter {
+	return buildFilter(true, filters...)
+}
+
+// DefaultMute creates a new handlerFilter that hides logs that match the given filters.
+// Later filters override earlier ones.
+func DefaultMute(filters ...Statement) Filter {
+	return buildFilter(false, filters...)
+}
+
+// ============================================================================
+// Filter Handler Types
+// ============================================================================
+
+type Filter func(ctx context.Context, lvl slog.Level) bool
+
+// FilterHandler interface that can be used with the log filtering system
+type FilterHandler interface {
+	logmods.Handler
+	// config parameter can be created with logfilter.DefaultShow(<filters...>) or logfilter.DefaultMute(<filters...>)
+	Set(filter Filter)
+}
+
+// filterHandler implements the Handler interface
+type filterHandler struct {
+	slog.Handler
+	filter atomic.Pointer[Filter]
+	attrs  map[string]slog.Value
+	group  string
+}
+
+var _ FilterHandler = &filterHandler{}
+
+func (f *filterHandler) Set(filter Filter) {
+	f.filter.Store(&filter)
+}
+
+func (f *filterHandler) Unwrap() slog.Handler {
+	return f.Handler
+}
+
+func (f *filterHandler) Enabled(ctx context.Context, lvl slog.Level) bool {
+	// First check if the log is enabled by the inner handler, because these are less expensive to evaluate
+	if !f.Handler.Enabled(ctx, lvl) {
+		return false
+	}
```

Comment:
The inner log handler is being set to filter everything at `Info` level, which will cause everything below it to be filtered, regardless of settings. I think we want to configure the test-logger to trace-level, or not respect the inner-handler here at all.

Reply:
I believe I fixed this, see latest push.